pypesto/sampling/theano.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+
+from ..objective import Objective
+from ..problem import Problem
+
+try:
+    import theano.tensor as tt
+    from theano.gof.null_type import NullType
+except ImportError:
+    tt = NullType = None
+
+
+class TheanoLogProbability(tt.Op):
+    """
+    Theano wrapper around a (non-normalized) log-probability function.
+
+    Parameters
+    ----------
+    problem:
+        The `pypesto.Problem` to analyze.
+    beta:
+        Inverse temperature (e.g. in parallel tempering).
+    """
+
+    itypes = [tt.dvector]  # expects a vector of parameter values when called
+    otypes = [tt.dscalar]  # outputs a single scalar value (the log prob)
+
+    def __init__(self, problem: Problem, beta: float = 1.):
+        self._objective: Objective = problem.objective
+
+        # initialize the log probability Op
+        self._log_prob = \
+            lambda x: - beta * self._objective(x, sensi_orders=(0,))
+
+        # initialize the sensitivity Op
+        if problem.objective.has_grad:
+            self._log_prob_grad = TheanoLogProbabilityGradient(problem, beta)
+        else:
+            self._log_prob_grad = None
+
+    def perform(self, node, inputs, outputs, params=None):
+        theta, = inputs
+        log_prob = self._log_prob(theta)
+        outputs[0][0] = np.array(log_prob)
+
+    def grad(self, inputs, g):
```

Comment:
Why not as docstrings?

Reply:
thx, will do when revising the code next time.